### PR TITLE
Design Change Proposal:  Use VisionComponent inside of VisionActor

### PR DESCRIPTION
**Pull request by Tim-Fronsee**
### Summary

In order to avoid having double logic (duplicated by me) between `VisionComponent` and `VisionActor`, this PR does the following:

- `VisionActor` now inherits from `AActor` 
- `VisionActor` contains a `VisionComponent` attached to its `RootComponent`
- Previous ROS vision related logic has been removed from `VisionActor` (which is now housed in `VisionComponent`)

Additional changes: 

- Adds missing includes to `VisionComponent`

While the ROS side of this PR (topic names, etc.) is unaffected by the change, the Unreal side may effect users inheriting from `VisionActor` directly.

Please let me know your thoughts,

Tim

### CODE REVIEW

**Comment by Sanic** [comment]
Hi @Tim-Fronsee 
That's a very good idea. On the VisionActor i would expect to see an automatic instantiation of a VisionComponent that will then be automatically attached to the VisionActor.
Right now one would have to assign it manually in the Editor to the UPROPERTY, right?

**Comment by Tim-Fronsee** [comment]
@Sanic Actually, that logic is inside the `VisionActor`'s constructor.

Using the `Spawn Actor from Class` BP node (please excuse the poorly cropped images), 

image

the `VisionComponent` is already attached the to the `RootComponent`,

image

File: Source/ROSIntegrationVision/Private/VisionActor.cpp [hunk 1]
```diff
@@ -2,910 +2,30 @@
...
-	//delete _depth_publisher;
-	//delete _tf;
-	UE_LOG(LogTemp, Display, TEXT("VisionActor got destroyed"));
+    
+    RootComponent = CreateDefaultSubobject<USceneComponent>(TEXT("Root"));
+    SetRootComponent(RootComponent);
+    
+    vision = CreateDefaultSubobject<UVisionComponent>(TEXT("Vision"));
+    vision->ParentLink = "/world";
+    vision->SetupAttachment(RootComponent);
```
Comment (Tim-Fronsee) [inline Source/ROSIntegrationVision/Private/VisionActor.cpp:18]: This is the line where the `VisionComponent` is attached to the `VisionActor`'s `RootComponent`.

**Comment by Sanic** [comment]
Thanks for the Clarification!

**Pull request merged**